src/pkg/cli/compose.go

```diff
@@ -107,15 +107,64 @@ func warnf(format string, args ...interface{}) {
 	HadWarnings = true
 }
 
-func loadCompose(filePath string, projectName string, overrideProjectName bool) (*compose.Project, error) {
-	// The default path for a Compose file is compose.yaml (preferred) or compose.yml that is placed in the working directory.
+func getComposeFilePath(composeDirAndFilePattern string) (string, error) {
+	filePattern := filepath.Base(composeDirAndFilePattern)
+	path := filepath.Dir(composeDirAndFilePattern)
+
+	// allow walking up the directory tree only if directory has not been specified
+	onlyLookInSpecificDir := strings.HasPrefix(composeDirAndFilePattern, path)
+
+	if !strings.HasSuffix(composeDirAndFilePattern, filePattern) {
+		return composeDirAndFilePattern, fmt.Errorf("no file or file pattern given")
+	}
+
+	// change directory to get compose file from
+	err := os.Chdir(path)
+	if err == nil {
+		// path may not be full path name (e.g. ../.. ) so get full path name for logs and if we need to walk the directory tree
+		path, err = os.Getwd()
+	}
+
+	// The Compose file is compose.yaml (preferred) or compose.yml that is placed in the current directory or higher.
 	// Compose also supports docker-compose.yaml and docker-compose.yml for backwards compatibility.
-	if files, _ := filepath.Glob(filePath); len(files) > 1 {
-		return nil, fmt.Errorf("multiple Compose files found: %q; use -f to specify which one to use", files)
-	} else if len(files) == 1 {
-		filePath = files[0]
+	// Users can override the file with another file pattern
+	if err == nil {
+		for {
+			Debug("Searching for ", filePattern, " in ", path)
+			if files, _ := filepath.Glob(filePattern); len(files) > 1 {
+				err = fmt.Errorf("multiple Compose files found: %q; use -f to specify which one to use", files)
+				break
+			} else if len(files) == 1 {
+				path = files[0]
+				break
+			}
+
+			if onlyLookInSpecificDir {
+				err = fmt.Errorf("no Compose file found in %q", path)
+				break
+			}
+
+			// compose file not found, try parent directory
+			nextPath := filepath.Dir(path)
+			if nextPath == path {
+				// we're reached the root folder, we're done
+				err = fmt.Errorf("no Compose file found")
+				break
+			}
+			path = nextPath
+			err = os.Chdir(path)
```

Comment:
`os.Chdir` may have unintended consequence of file lookup in other places to resolve to a different file than intended.

Reply:
I'll reset to the original on exit from the function.